examples/coap-client.c

```diff
@@ -1830,6 +1858,32 @@ main(int argc, char **argv) {
           obs_ms -= result;
         }
       }
+      if (repeat_count) {
+        /* Send off next request if appropriate */
+        repeat_ms -= result;
+        if (repeat_ms <= 0) {
```

Comment:
As this effectively is the same as checking for `repeat_ms <= result`, why not changing the type of `repeat_ms` to `uint32_t` and doing `repeat_ms -= result` only if `repeat_ms > result`?

Reply:
Made suggested change.  However, the code now has to read (as `result` is an `int`)
````
        /* Send off next request if appropriate */
        if (repeat_ms > (unsigned)result) {
          repeat_ms -= (unsigned)result;
        } else {
          /* Doing this once a second */
````
Also made use of the `ready `flag to not initiate the second request too early should DTLS setup take a long time.

Code updates pushed (but there are some generic build failures which I am investigating)

Reply:
Github have now fixed their build issues and code now builds as expected.